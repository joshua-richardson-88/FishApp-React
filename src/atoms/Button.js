import React from 'react';

export default function Button(props) {
  return (
    <button className='fish-button' onClick={props.action} data-testid='button'>
      {props.children}
    </button>
  );
}
