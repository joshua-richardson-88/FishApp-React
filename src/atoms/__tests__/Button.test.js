import React from 'react';
import Button from '../Button';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Atoms - Button Test', () => {
  const handleClick = jest.fn();

  it('matches the snapshot', () => {
    const { asFragment } = render(<Button action={handleClick}>Hello Bees!</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders text', () => {
    const { getByTestId } = render(<Button action={handleClick}>Hello Bees!</Button>);

    expect(getByTestId('button')).toHaveTextContent(/^Hello Bees!$/);
  });
  it('correctly handles clicks', () => {
    const { getByTestId } = render(<Button action={handleClick}>Hello Bees!</Button>);
    const button = getByTestId('button');

    userEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
