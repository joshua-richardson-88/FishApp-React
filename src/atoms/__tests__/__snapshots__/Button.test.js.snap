// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms - Button Test matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="fish-button"
    data-testid="button"
  >
    Hello Bees!
  </button>
</DocumentFragment>
`;
